File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.size() == 0) return "";
        if(s.size() == 1) return s;
        int len = INT_MIN, startIndex = 0;
        int count = 0;

        for(int i = 0; i<s.size() * 2 -1; i++){
            int l = i/2, r = (i+1)/2;
            while(l>=0 && r<s.size() && s[l] == s[r]){
                count = r - l + 1;
                if(count > len){
                    len = count;
                    startIndex = l;
                }
                l--;
                r++;
            }
            
        }
        return len == INT_MIN ? "" : s.substr(startIndex,len);
    }
};

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if(len(s) == 0): return ""
        if(len(s) == 1): return s
        max_len = 0
        ans = ""
        for i in range(1,len(s)*2 - 1):
            r = (i+1)//2
            l = i//2
            counter = 0
            while(l >= 0 and r < len(s) and s[l] == s[r]):
                    counter = r-l+1
                    if(counter > max_len):
                        max_len = counter
                        ans = s[l:r+1]
                    l -= 1
                    r += 1
        return ans